api/src/main/java/org/apache/iceberg/view/View.java

```diff
@@ -77,6 +77,15 @@ public interface View {
    */
   Map<String, String> properties();
 
+  /**
+   * Return the view's base location.
+   *
+   * @return this view's location
+   */
+  default String location() {
+    return null;
```

Comment:
I don't think we want to throw here

Reply:
A view would always have a location no? I would've thought we could follow the same pattern we have for `Table` and just have `BaseView` override and return the location (no default implementation needed, just a plain old interface method definition).

Reply:
this is only having a default implementation so that we don't break the API. Instead of returning null here we could just throw an exception like we do in other default methods

Reply:
Yeah, this is the unfortunate consequence of overlooking location when we added the APIs. We should definitely throw an exception. We never want Iceberg to return incorrect information.